src/Sylius/Behat/Context/Ui/Admin/ManagingProductsContext.php

```diff
@@ -205,6 +205,9 @@ public function theProductShouldAppearInTheShop($productName)
      */
     public function iWantToBrowseProducts()
     {
+        // Timeout because of elasticsearch update timing, use "curl 'localhost:9200/_cat/pending_tasks'" to check instead
+        sleep(1);
```

Comment:
Yes, but it's not related to the page itself and i can't rely on the display.

I think instead make an elasticsearchChecker (as the notificationChecker) and inject it into the context with a waitForPendingRequests().

WDYT ?